vital/tests/test_geo_polygon.cxx

```diff
@@ -0,0 +1,257 @@
+ */
+
+#include <test_common.h>
+
+#include <vital/types/geo_polygon.h>
+#include <vital/types/geodesy.h>
+#include <vital/types/polygon.h>
+#include <vital/plugin_loader/plugin_manager.h>
+
+#define TEST_ARGS ()
+
+DECLARE_TEST_MAP();
+
+// "It's a magical place." -- P.C.
+static auto const loc_ll = kwiver::vital::vector_2d{ -149.484444, -17.619482 };
+static auto const loc_utm = kwiver::vital::vector_2d{ 236363.98, 8050181.74 };
+
+static auto constexpr crs_ll = kwiver::vital::SRID::lat_lon_WGS84;
+static auto constexpr crs_utm_6s = kwiver::vital::SRID::UTM_WGS84_south + 6;
+
+// ----------------------------------------------------------------------------
+bool operator==( kwiver::vital::polygon const& a,
+                 kwiver::vital::polygon const& b )
+{
+  auto const k = a.num_vertices();
+  if ( b.num_vertices() != k )
+  {
+    return false;
+  }
+
+  for ( auto i = decltype(k){ 0 }; i < k; ++i )
+  {
+    if ( a.at( i ) != b.at( i ) )
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+bool operator!=( kwiver::vital::polygon const& a,
+                 kwiver::vital::polygon const& b )
+{
+  return !( a == b );
+}
+
+// ----------------------------------------------------------------------------
+int
+main(int argc, char* argv[])
+{
+  CHECK_ARGS(1);
+  kwiver::vital::plugin_manager::instance().load_all_plugins();
+
+  testname_t const testname = argv[1];
+
+  RUN_TEST(testname);
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(default_constructor)
+{
+  kwiver::vital::geo_polygon p;
+
+  if ( ! p.is_empty() )
+  {
+    TEST_ERROR("The default polygon is not empty");
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(constructor_point)
+{
+  kwiver::vital::geo_polygon p{ { loc_ll }, crs_ll };
+
+  if ( p.is_empty() )
+  {
+    TEST_ERROR("The constructed polygon is empty");
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(assignment)
+{
+  kwiver::vital::geo_polygon p;
+  kwiver::vital::geo_polygon const p1{ { loc_ll }, crs_ll };
+  kwiver::vital::geo_polygon const p2;
+
+  if ( ! p.is_empty() )
+  {
+    TEST_ERROR("The default polygon is not empty");
+  }
+
+  p = p1;
+
+  if ( p.is_empty() )
+  {
+    TEST_ERROR("The polygon is empty after assignment from non-empty polygon");
+  }
+
+  if ( p.polygon() != p1.polygon() )
+  {
+    TEST_ERROR("The polygon has the wrong location after assignment");
+  }
+
+  if ( p.crs() != p1.crs() )
+  {
+    TEST_ERROR("The polygon has the wrong CRS after assignment");
+  }
+
+  p = p2;
+
+  if ( ! p.is_empty() )
+  {
+    TEST_ERROR("The polygon is not empty after assignment from polygon point");
+  }
+}
+
+// ----------------------------------------------------------------------------
+IMPLEMENT_TEST(api)
+{
+  /*
```

Comment:
Why are all of these tests commented out?  I would either enable them or remove them entirely.

Reply:
This file was copied from `test_geo_point.cxx`. Converting these from working on a `geo_point` to working on a `geo_polygon` was the point at which doing it well became... awkward with sprokit's test framework (i.e. the point at which I said "screw it, I'm going to write this with Google Test").